src/provider/model/Model.ts

```diff
@@ -29,6 +30,7 @@ export class Model {
 
     private _windows: AppWindow[];
     private _channels: ContextChannel[];
+    private windowAddedSignal: Signal0 = new Signal0();
```

Comment:
We're using onFooHappened, or just fooHappened elsewhere (which is an inconsistency that should be resolved in itself, but let's stick to not postfixing 'Signal')

Reply:
done